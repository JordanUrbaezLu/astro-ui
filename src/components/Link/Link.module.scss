@use "../../utility/globals";

.linkContainer {
  position: relative;
  text-decoration: none;

  &.primary::after {
    background-color: globals.$primary-text-color;
    bottom: 0;
    content: "";
    height: 1.5px;
    left: 0;
    position: absolute;
    transform: scaleX(0);
    transform-origin: bottom right;
    transition: transform 0.25s ease-out;
    width: 100%;
  }

  &.secondary::after {
    background-color: globals.$secondary-text-color;
    bottom: 0;
    content: "";
    height: 1.5px;
    left: 0;
    position: absolute;
    transform: scaleX(0);
    transform-origin: bottom right;
    transition: transform 0.25s ease-out;
    width: 100%;
  }

  &:hover::after {
    transform: scaleX(1);
    transform-origin: bottom left;
  }
}
