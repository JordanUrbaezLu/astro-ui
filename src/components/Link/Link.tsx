import classNames from "classnames";
import * as React from "react";
import { Typography } from "../Typography/Typography";
import styles from "./Link.module.scss";

export type LinkVariant = "primary" | "secondary";

export interface LinkProps
  extends React.ComponentPropsWithoutRef<"a"> {
  /**
   * The content for the Link
   */
  children: string;
  /**
   * The href for the Link
   */
  href: string;
  /**
   * The variant for the Link
   *
   * @default "primary"
   */
  variant?: LinkVariant;
}

/**
 * @public
 *
 * @description
 *
 * The Link component allows a user to navigate to a new page.
 */
export const Link: React.FC<LinkProps> = (props) => {
  const {
    className,
    children,
    href,
    variant = "primary",
    ...rest
  } = props;

  return (
    <a
      className={classNames(
        className,
        styles.linkContainer,
        variant === "primary" ? styles.primary : styles.secondary
      )}
      href={href}
      {...rest}
    >
      <Typography type="h2" variant={variant}>
        {children}
      </Typography>
    </a>
  );
};

Link.displayName = "Link";
