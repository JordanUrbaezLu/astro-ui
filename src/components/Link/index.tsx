/* istanbul ignore file */
export * from "./Link";
