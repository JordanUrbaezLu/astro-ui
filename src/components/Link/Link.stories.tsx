import { FlexRowContainer } from "../../../.storybook/components/FlexRowContainer";
import { Link } from "./Link";

export default {
  component: Link,
  title: "Components/Navigation/Link",
};

export const Default = () => (
  <div
    style={{
      background: "rgb(60 60 60)",
      padding: "6px",
      width: "fit-content",
    }}
  >
    <Link href="#">Default</Link>
  </div>
);

export const Variant = () => {
  return (
    <FlexRowContainer>
      <div style={{ background: "rgb(60 60 60)", padding: "6px" }}>
        <Link href="#" variant="primary">
          Primary
        </Link>
      </div>
      <div>
        <Link href="#" variant="secondary">
          Secondary
        </Link>
      </div>
    </FlexRowContainer>
  );
};
