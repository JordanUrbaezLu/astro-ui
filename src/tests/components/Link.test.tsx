import { render, screen } from "@testing-library/react";
import { Link, LinkVariant } from "../../components/Link/Link";
import { axe } from "jest-axe";

describe("Link", () => {
  test("Should render component correctly.", () => {
    const { container } = render(
      <Link href="https://www.google.com/">Google.com</Link>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(
      <Link href="https://www.google.com/">Google.com</Link>
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test.each<LinkVariant>(["primary", "secondary"])(
    "Should render variants correctly.",
    (variant) => {
      render(
        <Link href="https://www.google.com/" variant={variant}>
          Google.com
        </Link>
      );
      expect(screen.getAllByRole("link")[0]).toHaveClass(variant);
    }
  );
});
