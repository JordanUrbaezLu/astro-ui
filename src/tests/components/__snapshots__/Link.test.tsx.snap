// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Should render component correctly. 1`] = `
<a
  class="linkContainer primary"
  href="https://www.google.com/"
>
  <div
    class="typography typographyh2 primary normal"
  >
    Google.com
  </div>
</a>
`;
